storage.js

```diff
@@ -0,0 +1,21 @@
+let formData = {
+  fullName: '',
+  email: '',
+  message: '',
+};
+
+const form = document.querySelector('form');
+
+if (localStorage.getItem('formData') != null) {
+  formData = JSON.parse(localStorage.getItem('formData'));
+  form.name.value = formData.fullName;
+  form.email.value = formData.email;
+  form.message.value = formData.message;
+}
+
+form.addEventListener('submit', () => {
+  formData.fullName = form.name.value;
+  formData.email = form.email.value;
+  formData.message = form.message.value;
+  localStorage.setItem('formData', JSON.stringify(formData));
+});
```

Comment:
- [ ] Great job implementing local storage registry. But according to the requirements, the inputs should be stored when the user does not submit the form, so whenever he reloads the page, all the data is back in the inputs, so you will need the following:
1. [REQUIRED] When the user changes the content of any input field, the whole object (in your case `form-data`) is updated and saved to the local storage - one good option would be the `onChange event`, you can read more about it [here](https://www.w3schools.com/jsref/event_onchange.asp).
2. When the user loads the page, the code checks if there are any saved data in the local storage and if so, the input fields are pre-filled with that data.
- [ ] [OPTIONAL]: The way you implemented the second point should work fine, but maybe you should consider adding a `window.onload` function wrapping it, you can read more about it [here](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onload).

Reply:
Thank you for your review.  I would like additional information. 

If I use **the "onchange event" the function is called only when I change my focus of the input** but if I use the **"keyup event" the function is called at each change of letter and therefore I can save the data in real-time.** 

What is the best event to use?